app/models/message.rb

```diff
@@ -59,4 +62,15 @@ def chat_message_link
       anchor: "chat-row-#{id}"
     )
   end
+
+  private
+
+  def notify_recipient
+    if reply?
+      MessageReceived.with(contributor_id: sender.id, request_id: request.id, message_id: id).deliver_later(User.all)
+    elsif !broadcasted?
+      ChatMessageSent.with(contributor_id: recipient.id, request_id: request.id, user_id: Current.user.id,
+                           message_id: id).deliver_later(User.all)
```

Comment:
Also commented on a [PR adding some improvements](https://github.com/tactilenews/100eyes/pull/1431#discussion_r958347111), which has been merged into this one:

> @tillprochaska maybe we can discuss this? I found it quite surprising that we don't save any information on which person in the newsroom has sent a "chat messge" to a contributor.

> I understand that we save everything as a Message and then we do a slightly complicated way of deciding if it's a broadcasted?,reply? or a manually_created? message, but we don't save any information on which User sent the message.

> I guess we could not save it in the ActivityNotification as well, then we could completely remove the ActiveSupport::CurrentAttributes. At present, we use this information to display which User sent out the chat message.

Haven't received a response on this issue.

Reply:
This is for historical reasons, as the prototype didn't have multi-user auth. Storing the user makes sense!

Reply:
See https://github.com/tactilenews/100eyes/issues/1456